### PR TITLE
[ES] Spanish translation week 12, section 3

### PR DESCRIPTION
**Pull request by zxul767**
While running locally to test the changes, I found and fixed a few broken things (e.g., the French translation for week 6 had an error that prevented the build from completing at all)

I also added some scaffolding to be completed by others (myself included) in future PRs (I think it's best to avoid too much WIP.)

### CODE REVIEW

File: docs/es/week09/09-1.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (zxul767) [inline docs/es/week09/09-1.md:1]: This change wasn't strictly necessary for this PR, but I added it so that the numbers matched in the local preview.

File: docs/fr/week06/06-2.md [hunk 2]
```diff
@@ -492,7 +492,7 @@ L'idée d'un réseau mémoire est qu'il y a deux parties importantes dans votre
 Pour un réseau mémoire, il y a une entrée au réseau, $x$ (pensez à cela comme une adresse de la mémoire), et comparez ce $x$ avec les vecteurs $k_1, k_2, k_3, \cdots$ ("clés") à travers un produit scalaire. En les faisant passer par une softmax, on obtient un tableau de nombres dont la somme est égale à un. Et il y a un ensemble d'autres vecteurs $v_1, v_2, v_3, \cdots$ ("valeurs"). Multipliez ces vecteurs par les scalaires provenant du softmax et additionnez ces vecteurs vous donne le résultat (notez la ressemblance avec le mécanisme d'attention).
 
 <center>
-<img src="{{site.baseurl/images/week06/06-2/MemoryNetwork1.png" height="300px"/><br>
+<img src="{{site.baseurl}}/images/week06/06-2/MemoryNetwork1.png" height="300px"/><br>
```
Comment (zxul767) [inline docs/fr/week06/06-2.md:495]: This was the error preventing the build from completing locally.

File: docs/es/week12/12-3.md [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+lang: es
+translator: Willebaldo Gómez Jiménez
+translation-date: 14 Sep 2020
+---
+
+<!-- ## [Attention](https://www.youtube.com/watch?v=f01J0Dri-6k&t=69s) -->
+## [Atención](https://www.youtube.com/watch?v=f01J0Dri-6k&t=69s)
+
+<!-- We introduce the concept of attention before talking about the Transformer architecture. There are two main types of attention: self attention *vs.* cross attention, within those categories, we can have hard *vs.* soft attention. -->
+Presentamos el concepto de "atención" antes de hablar sobre la arquitectura del Transformador. Existen dos tipos principales de atención: auto atención y atención cruzada. Dentro de esas categorías, distinguimos entre atención "ténue" y atención "intensa".
```
Comment (zxul767) [inline docs/es/week12/12-3.md:16]: I used the terms "atención ténue" and "atención intensa" for "soft attention" and "hard attention", respectively, but I still have doubts about whether we should just use the English terms. Thoughts?

File: docs/es/week12/12-3.md [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+
+<!-- Depending on the constraints we impose on the vector $\vect{a}$, we can achieve hard or soft attention. -->
+Dependiendo de las restricciones que impongamos en el vector $\vect{a}$, podemos conseguir atención intensa o atención ténue.
+
+
+<!-- #### Hard Attention -->
+#### Atención Intensa
+
+<!-- With hard-attention, we impose the following constraint on the alphas: $\Vert\vect{a}\Vert_0 = 1$. This means $\vect{a}$ is a one-hot vector. Therefore, all but one of the coefficients in the linear combination of the inputs equals zero, and the hidden representation reduces to the input $\boldsymbol{x}_i$ corresponding to the element $\alpha_i=1$. -->
+En la atención intensa, imponemos las siguientes restricciones en las alfas: $\Vert\vect{a}\Vert_0 = 1$. Esto significa que $\vect{a}$ es un vector con codificación "one-hot". Por lo tanto, todos los coeficientes (con excepción de uno) son iguales a cero en la combinación lineal de las entradas, y la representación interna se reduce a la entrada $\boldsymbol{x}_i$ que corresponde al elemento $\alpha_i=1$.
```
Comment (zxul767) [inline docs/es/week12/12-3.md:63]: Is there a better translation for "one-hot encoding" in Spanish?

File: docs/es/week12/12-3.md [hunk 5]
```diff
@@ -0,0 +1,546 @@
...
+<!-- Since each hidden state is a linear combination of the inputs $\boldsymbol{X}$ and a vector $\vect{a}$, we obtain a set of $t$ hidden states, which we can stack into a matrix $\boldsymbol{H}\in \mathbb{R}^{n \times t}$. -->
+Puesto que cada estado oculto es una combinación lineal de las entradas $\boldsymbol{X}$ y un vector $\vect{a}$, obtenemos un conjunto $t$ de estados ocultos, los cuales podemos apilar en una matriz $\boldsymbol{H}\in \mathbb{R}^{n \times t}$.
+
+$$
+\boldsymbol{H}=\boldsymbol{XA}
+$$
+
+
+<!-- ## [Key-value store](https://www.youtube.com/watch?v=f01J0Dri-6k&t=1056s) -->
+## [Almacén de clave/valor](https://www.youtube.com/watch?v=f01J0Dri-6k&t=1056s)
```
Comment (zxul767) [inline docs/es/week12/12-3.md:107]: I'm not convinced about the translation for "key-value store" that I used here. Suggestions?

File: docs/es/week12/12-3.md (hunk 3 above)
Comment (Atcold) [reply docs/es/week12/12-3.md:16]: In this context, soft means "smooth", hard means "1-hot". I think *suave* and *duro* are better choices. What do you think?

File: docs/fr/week06/06-2.md (hunk 2 above)
Comment (Atcold) [reply docs/fr/week06/06-2.md:495]: Yeah, we've fixed it already.

File: docs/es/week12/12-3.md (hunk 4 above)
Comment (Atcold) [reply docs/es/week12/12-3.md:63]: "uno-caliente"?

File: docs/es/week12/12-3.md (hunk 5 above)
Comment (Atcold) [reply docs/es/week12/12-3.md:107]: I think you may want to discuss on Slack. I'm not a native speaker.

File: docs/_config.yml [hunk 6]
```diff
@@ -280,12 +280,23 @@ es:
       sections:
         - path: es/week08/08-1.md
         - path: es/week08/08-2.md
+    - path: es/week09/09.md
+      sections:
+        - path: es/week09/09-1.md
+        - path: es/week09/09-2.md
+        - path: es/week09/09-3.md
```
Comment (Atcold) [inline docs/_config.yml:287]: We don't have the translation of week 9. Why did you add it?

File: docs/es/week09/09-1.md (hunk 1 above)
Comment (Atcold) [reply docs/es/week09/09-1.md:1]: This stuff is going to create merge conflicts. Please, remove these boilerplates.

**Review by Atcold — CHANGES_REQUESTED** [review]

File: docs/es/week12/12-3.md (hunk 5 above)
Comment (xcastilla) [reply docs/es/week12/12-3.md:107]: I've never heard anyone saying it in Spanish, but "Almacén de clave/valor" or "Base de datos clave/valor" seems to be right.
For reference:
[https://aws.amazon.com/es/nosql/key-value/](https://aws.amazon.com/es/nosql/key-value/)

File: docs/es/week12/12-3.md [hunk 7]
```diff
@@ -0,0 +1,546 @@
...
+
+<!-- A key-value store is a paradigm designed for storing (saving), retrieving (querying) and managing associative arrays (dictionaries / hash tables). -->
+Un almacén de clave/valor es un paradigman diseñado para almacenar (guardar), recuperar (consultar) y manejar arreglos asociativos (dicccionarios / tablas de hash.)
```
Comment (xcastilla) [inline docs/es/week12/12-3.md:110]:
```suggestion
Un almacén de clave/valor es un paradigma diseñado para almacenar (guardar), recuperar (consultar) y manejar arreglos asociativos (dicccionarios / tablas de hash.)
```

File: docs/es/week12/12-3.md [hunk 8]
```diff
@@ -0,0 +1,546 @@
...
+Sin embargo, aún es posible transformar los valores multi-cabeza para que tengan la dimensión original $\R^d$ usando un $\vect{W_h} \in \mathbb{R}^{d \times hd}$. Esta es sólo una de las posibles maneras de implementar el almacén de clave/valor.
+
+
+<!-- ## [The Transformer](https://www.youtube.com/watch?v=f01J0Dri-6k&t=2114s) -->
+## [El Transformador](https://www.youtube.com/watch?v=f01J0Dri-6k&t=2114s)
+
+<!-- Expanding on our knowledge of attention in particular, we now interpret the fundamental building blocks of the transformer. In particular, we will take a forward pass through a basic transformer, and see how attention is used in the standard encoder-decoder paradigm and compares to the sequential architectures of RNNs. -->
+Con el fin de expandir nuestro conocimiento sobre atención, interpretaremos ahora los bloques fundamentales del transformador. En particular, haremos un recorrido de principio a fin de un transformador básico, y veremos cómo se usa la atención en el paradigma estándar del codificador-decodificador, y cómo se compara esto con las arquitecturas secuenciales de las RNRs (Redes Neuronales Recurrentes).
```
Comment (xcastilla) [inline docs/es/week12/12-3.md:250]: I have a question here. I've seen in the wiki for other languages that acronyms shouldn't be changed:
e.g. [https://github.com/Atcold/pytorch-Deep-Learning/wiki/Italian-translation#rules](https://github.com/Atcold/pytorch-Deep-Learning/wiki/Italian-translation#rules)
Are we following this rule in the Spanish translation as well?

File: docs/es/week12/12-3.md [hunk 9]
```diff
@@ -0,0 +1,546 @@
...
+
+<center>
+<img src="{{site.baseurl}}/images/week12/12-3/figure1.png" style="zoom: 60%; background-color:#DCDCDC;" /><br>
+<!-- <b>Figure 1:</b> Two example diagrams of an autoencoder. The model on the left shows how an autoencoder can be design with two affine transformations + activations, where the image on the right replaces this single "layer" with an arbitrary module of operations. -->
+<b>Figura 1:</b> Dos diagramas ejemplificando un auto-codificador. El modelo de la izquierda muestra cómo un auto-codificador se puede diseñar con dos transformaciones afines + activaciones, mientras que el modelo de la derecha reemplaza esta "capa" única con un módulo arbitrario de operaciones.
```
Comment (xcastilla) [inline docs/es/week12/12-3.md:262]:
```suggestion
<b>Figura 1:</b> Dos diagramas ejemplificando un autoencoder. El modelo de la izquierda muestra cómo un auto-codificador se puede diseñar con dos transformaciones afines + activaciones, mientras que el modelo de la derecha reemplaza esta "capa" única con un módulo arbitrario de operaciones.
```
I saw in the translation for [lesson 7-3](https://github.com/Atcold/pytorch-Deep-Learning/blob/master/docs/es/week07/07-3.md) that _autoencoder_ was not translated to Spanish.  
We should either review lesson 7 or use the English name here to keep the notation consistent. What do you think?

File: docs/es/week12/12-3.md [hunk 10]
```diff
@@ -0,0 +1,546 @@
...
+
+<!-- There are a few important facts we left out before to explain the most important modules of a transformer, but will need to discuss them now to understand how transformers can achieve state-of-the-art results in language tasks. -->
+Hay algunos pocos hechos importantes que dejamos de lado al explicar los módulos más importantes del transformador, pero necesitamos discutirlos ahora para entender cómo los transformadores pueden lograr resultados de vanguardia en tareas de procesamiento de lenguaje.
+
+
+<!-- #### Positional encoding -->
+#### Codificación Posicional
+
+<!-- Attention mechanisms allow us to parallelize the operations and greatly accelerate a model's training time,  but loses sequential information. The positional encoding feature enables allows us to capture this context. -->
+Los mecanismos de atención nos permiten ejecutar en paralelo las operaciones y acelerar enormemente el tiempo de entrenamiento, pero al hacerlo exista una pérdida de información secuencial. La posibilidad de aAgregar codificación posicional nos permite evitar perder este contexto.
```
Comment (xcastilla) [inline docs/es/week12/12-3.md:346]:
```suggestion
Los mecanismos de atención nos permiten ejecutar en paralelo las operaciones y acelerar enormemente el tiempo de entrenamiento, pero al hacerlo exista una pérdida de información secuencial. La posibilidad de agregar codificación posicional nos permite evitar perder este contexto.
```

File: docs/es/week12/12-3.md [hunk 11]
```diff
@@ -0,0 +1,546 @@
...
+### Modelos de Lenguaje basados en Palabras
+
+<!-- There are a few important facts we left out before to explain the most important modules of a transformer, but will need to discuss them now to understand how transformers can achieve state-of-the-art results in language tasks. -->
+Hay algunos pocos hechos importantes que dejamos de lado al explicar los módulos más importantes del transformador, pero necesitamos discutirlos ahora para entender cómo los transformadores pueden lograr resultados de vanguardia en tareas de procesamiento de lenguaje.
+
+
+<!-- #### Positional encoding -->
+#### Codificación Posicional
+
+<!-- Attention mechanisms allow us to parallelize the operations and greatly accelerate a model's training time,  but loses sequential information. The positional encoding feature enables allows us to capture this context. -->
```
Comment (xcastilla) [inline docs/es/week12/12-3.md:345]:
```suggestion
<!-- Attention mechanisms allow us to parallelize the operations and greatly accelerate a model's training time,  but loses sequential information. The positional encoding feature allows us to capture this context. -->
```
I think we should have either _enables_ or _allows_ here, but not both @Atcold

File: docs/es/week12/12-3.md (hunk 4 above)
Comment (xcastilla) [reply docs/es/week12/12-3.md:63]: I'd personally leave it as "one-hot". When I was taking CS back at college I remember having seen it that way.

Comment (zxul767) [reply docs/es/week12/12-3.md:63]: I agree with @xcastilla, the term "uno-caliente" sounds very strange in Spanish.

File: docs/es/week12/12-3.md (hunk 3 above)
Comment (zxul767) [reply docs/es/week12/12-3.md:16]: Makes sense. Will push in a bit a change for this.

File: docs/es/week09/09-1.md (hunk 1 above)
Comment (zxul767) [reply docs/es/week09/09-1.md]: Done

File: docs/_config.yml (hunk 6 above)
Comment (zxul767) [reply docs/_config.yml]: I was just thinking of adding scaffolding, but you're right, it's likely to cause merge conflicts, so I removed it.

File: docs/es/week12/12-3.md (hunk 8 above)
Comment (zxul767) [reply docs/es/week12/12-3.md:250]: I wasn't aware of that rule. Thanks for bringing it up. I think it makes sense to use it for consistency with other translations.

File: docs/es/week12/12-3.md (hunk 9 above)
Comment (zxul767) [reply docs/es/week12/12-3.md:262]: In this case, I feel like "auto-codificador" is a good choice (unless we agreed to never translate names of technical terms like that). Maybe a good discussion to have in Slack.

File: docs/es/week12/12-3.md (hunk 4 above)
Comment (Atcold) [reply docs/es/week12/12-3.md]: Fine with me. I'm not a native speaker.

**Review by Atcold — CHANGES_REQUESTED** [review]
Please, don't change the website generation version.
You're going to break things this way.
Please, revert this edit.

image

**Comment by Atcold** [comment]
@zxul767 and @xcastilla can you two help me with the administration of the Spanish translating group?
You seem the most technically skilled, and I'm going a little crazy with the overhead of handling 300 people alone.

Thanks so much!

File: docs/es/week12/12-3.md (hunk 9 above)
Comment (xcastilla) [reply docs/es/week12/12-3.md:262]: Sounds good to me!

**Comment by zxul767** [comment]
> Please, don't change the website generation version.
> You're going to break things this way.
> Please, revert this edit.
Done!

**Review by GastonMazzei — APPROVED** [review]

**Review by xcastilla — APPROVED** [review]

**Review by xcastilla — APPROVED** [review]

**Review by Atcold — APPROVED** [review]
Fantastic!
